src/nupic/types/Sdr.cpp

```diff
@@ -323,6 +323,18 @@ namespace sdr {
     }
 
 
+    void SparseDistributedRepresentation::intersection(std::vector<const SDR*> inputs) {
```

Comment:
> std::vector<const SDR*> inputs 

do SDRs need to be pointers? Can this be
`vector<const SDR>& input` ? 

Or at least,
`vector<const SDR*>& input`

Reply:
C++ does not allow references inside of vectors.

I will change it to: `vector<const SDR*>& input`